Nightly Build Failure 2024-07-24

# Overview

What do you think happened, if anything?
We most likely forgot to update some validation tests for the heat rate tables when integrating new EIA 923 data yesterday #3677.

# Next steps

What next steps do we need to do to understand or remediate the issue?
Review the changed tables and make sure that the new rows of data are as expected. Then, update the expected row counts.

# Verify that everything is fixed!

Once you've applied any necessary fixes, make sure that the nightly build outputs are all in their right places.

```[tasklist]
- [ ] [S3 distribution bucket](https://s3.console.aws.amazon.com/s3/buckets/pudl.catalyst.coop?region=us-west-2&bucketType=general&prefix=nightly/&showversions=false) was updated at the expected time
- [ ] [GCP distribution bucket](https://console.cloud.google.com/storage/browser/pudl.catalyst.coop/nightly;tab=objects?project=catalyst-cooperative-pudl) was updated at the expected time
- [ ] [GCP internal bucket](https://console.cloud.google.com/storage/browser/builds.catalyst.coop) was updated at the expected time
- [ ] [Datasette PUDL version](https://data.catalyst.coop/pudl/core_pudl__codes_datasources) points at the same hash as [nightly](https://github.com/catalyst-cooperative/pudl/tree/nightly)
- [ ] [Zenodo sandbox record](https://sandbox.zenodo.org/doi/10.5072/zenodo.5563) was updated to the record number in the logs (search for `zenodo_data_release.py` and `Draft` in the logs, to see what the new record number should be!)
```

# Relevant logs
[link to build logs from internal distribution bucket](https://console.cloud.google.com/storage/browser/_details/builds.catalyst.coop/2024-07-24-0603-4cba9f5bf-main/2024-07-24-0603-4cba9f5bf-main-pudl-etl.log;tab=live_object)

```
FAILED test/validate/mcoe_test.py::test_minmax_rows_mcoe[eia_annual-hr_by_unit-406067-33946]
FAILED test/validate/mcoe_test.py::test_minmax_rows_mcoe[eia_annual-hr_by_gen-634378-52978]
FAILED test/validate/mcoe_test.py::test_minmax_rows_mcoe[eia_annual-fuel_cost-634378-52978]
FAILED test/validate/mcoe_test.py::test_minmax_rows_mcoe[eia_annual-capacity_factor-5494698-459691]
FAILED test/validate/mcoe_test.py::test_minmax_rows_mcoe[eia_annual-mcoe-5495136-459726]
FAILED test/validate/mcoe_test.py::test_minmax_rows_mcoe[eia_monthly-hr_by_unit-406067-33946]
FAILED test/validate/mcoe_test.py::test_minmax_rows_mcoe[eia_monthly-hr_by_gen-634378-52978]
FAILED test/validate/mcoe_test.py::test_minmax_rows_mcoe[eia_monthly-fuel_cost-634378-52978]
FAILED test/validate/mcoe_test.py::test_minmax_rows_mcoe[eia_monthly-capacity_factor-5494698-459691]
FAILED test/validate/mcoe_test.py::test_minmax_rows_mcoe[eia_monthly-mcoe-5495136-459726]
```

--- a/test/validate/mcoe_test.py
+++ b/test/validate/mcoe_test.py
@@ -94,11 +94,11 @@ def test_no_null_rows_mcoe(pudl_out_mcoe, live_dbs, df_name, thresh):
 @pytest.mark.parametrize(
     "df_name,monthly_rows,annual_rows",
     [
-        ("hr_by_unit", 406_067, 33_946),
-        ("hr_by_gen", 634_378, 52_978),
-        ("fuel_cost", 634_378, 52_978),
-        ("capacity_factor", 5_494_698, 459_691),
-        ("mcoe", 5_495_136, 459_726),
+        ("hr_by_unit", 409_787, 34_258),
+        ("hr_by_gen", 640_960, 53_530),
+        ("fuel_cost", 640_960, 53_530),
+        ("capacity_factor", 5_494_932, 459_711),
+        ("mcoe", 5_495_370, 459_746),
     ],
 )
 def test_minmax_rows_mcoe(pudl_out_mcoe, live_dbs, monthly_rows, annual_rows, df_name):

